internal/runtime/client/client.go

```diff
@@ -382,6 +386,23 @@ func (c *client) CallExtension(ctx context.Context, hook runtimecatalog.Hook, fo
 	return nil
 }
 
+// prepareRequest creates a new request object by adding the appropriate values from the registration.
+func prepareRequest(request runtimehooksv1.RequestObject, registration *runtimeregistry.ExtensionRegistration) runtimehooksv1.RequestObject {
+	// Merge the settings from registration with the settings in the request.
+	// The values in request take precedence over the values in the registration.
+	// Create a deepcopy object to avoid side-effects on the request object.
+	request = request.DeepCopyObject().(runtimehooksv1.RequestObject)
+	settings := registration.Settings
+	if settings == nil {
```

Comment:
Let's create a new map and then add key/value pairs from both settings.

The current code will modify the map inside of the registration which means that settings of all calls will add up in the registration

Reply:
Let's make sure we cover this edge case with a unit test
(we had this bug in various parts of our case base already)